src/features/locators/components/LocatorEditDialog.tsx

```diff
@@ -93,7 +118,70 @@ export const LocatorEditDialog: React.FC<Props> = ({
     setIsEditedName(true);
   };
 
-  const handleOk = () => {
+  const handleCreateCustomLocator = async () => {
+    let newLocator: Locator = {
+      ...newLocatorStub,
+      pageObj: pageObjectId,
+      isCustomName: isEditedName,
+      locatorType: formLocatorType,
+    };
+
+    const isLocatorFieldTouched = form.isFieldTouched("locator");
+
+    await form
+      .validateFields()
+      .then(({ name, type, locator }) => {
+        newLocator = {
+          ...newLocator,
+          locator: { ...newLocator.locator, customXpath: locator },
+          predicted_label: type.toLowerCase(),
+          // in case if user didn't touch locator field to avoid forceUpdate
+          message: isLocatorFieldTouched ? validationMessage : LocatorValidationWarnings.NotFound,
+          name,
+          type,
+        };
+      })
+      .catch((err) => console.log(err));
+
+    switch (getLocatorValidationStatus(validationMessage)) {
+      case ValidationStatus.WARNING:
+        newLocator.element_id = `${generateId()}_${pageObjectId}`;
+        break;
+      case ValidationStatus.ERROR:
+        return;
```

Comment:
what is the purpose of this `case`? can it be omitted?

Reply:
I believe yes, we can